src/eigen/preconditioners.jl

```diff
@@ -33,7 +33,13 @@ mutable struct PreconditionerTPA{T <: Real}
 end
 
 function PreconditionerTPA(basis::PlaneWaveBasis{T}, kpt::Kpoint; default_shift=1) where T
-    scaling = only([t for t in basis.model.term_types if t isa Kinetic]).scaling_factor
+    kinetic_term = [t for t in basis.model.term_types if t isa Kinetic]
+    if length(kinetic_term) >= 1
+        scaling = only(kinetic_term).scaling_factor
+    else
+        # If user do strange computations without a kinetic term.
+        scaling = 1
+    end
```

Comment:
This is a type instability. But here just raise an error; the preconditioner is not relevant in this case, users should disable preconditioning. So just `kinetic_terms = []; length() >= 1 || error(); scaling = only(kinetic_terms).scaling`